nion/swift/ExportDialog.py

```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
-
+        self.width_line_edit = u.create_line_edit(
+            text="initial_width",
+            on_text_edited="value_updated",
+            background_color="white",
+            name="width_value_line_edit"
+        )
+        self.height_line_edit = u.create_line_edit(
+            text="initial_height",
+            on_text_edited="value_updated",
+            background_color="white",
+            name="height_value_line_edit"
+        )
+        width_row = u.create_row(
+            u.create_label(text=_("Width:"), width=80),
+            self.width_line_edit,
+            spacing=12
+        )
+        height_row = u.create_row(
+            u.create_label(text=_("Height:"), width=80),
+            self.height_line_edit,
+            spacing=12
+        )
+        # Combo box for units
+        self.units_combo_box = u.create_combo_box(
+            items=[_("Inches"), _("Centimeters"), _("Pixels")],
+            current_index="@binding(unit_model.value)",
+            on_current_index_changed="unit_changed"
+        )
+        units_row = u.create_row(
+            u.create_label(text=_("Units:"), width=80),
+            self.units_combo_box,
+            spacing=12
+        )
+        main_page = u.create_column(width_row, height_row, units_row, spacing=12, margin=12)
         self.ui_view = main_page
 
+    def value_updated(self, widget: UserInterface.LineEditWidget, text: str) -> None:
+        try:
+            if widget == self.width_value_line_edit:
+                new_width = float(text)
+                new_height = new_width / self.aspect_ratio
+                if self.unit_model == UnitType.PIXELS:
+                    new_height = round(new_height)
+                self.height_value_line_edit.text = str(new_height)
+                self.height_value_line_edit.background_color = "#lightgrey"
+                self.width_value_line_edit.background_color = "white"
+            elif widget == self.height_value_line_edit:
+                new_height = float(text)
+                new_width = new_height * self.aspect_ratio
+                if self.unit_model == UnitType.PIXELS:
+                    new_width = round(new_width)
+                self.width_value_line_edit.text = str(new_width)
+                self.width_value_line_edit.background_color = "#lightgrey"
+                self.height_value_line_edit.background_color = "white"
+        except ValueError:
+            pass  # Handle the case where the input is not a valid integer
+
+    def unit_changed(self, widget: UserInterface.ComboBoxWidget, current_index: int) -> None:
+        new_unit_type = UnitType(current_index)
+        self.size = self.convert_to_pixels(
+            Geometry.FloatSize(height=float(self.height_value_line_edit.text or 0),
+                               width=float(self.width_value_line_edit.text or 0)),
+            self.previous_unit_index
+        )
+        new_width, new_height = self.convert_from_pixels(self.size, new_unit_type)
```

Comment:
This line is wrong:
`new_width, new_height = self.convert_from_pixels(self.size, new_unit_type)`

It should be `new_shape = self.convert_from_pixels(self.size, new_unit_type)` and then use `new_shape.width` and `new_shape.height`.

In the incorrect code, you assign the `FloatSize` returned by `self.convert_from_pixels` to two variables and a Python language feature called 'unpacking' occurs where it assigns the return value subscripted by each unpacked value into the variables. So if the result were in `r`, `new_width` would get `r[0]` and `new_height` would get `r[1]`. This is wrong since index `0` is for height. A quick test of the dialog with the line plot reveals this bug.